enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -2365,16 +2367,17 @@ const AugmentedReturn &EnzymeLogic::CreateAugmentedPrimal(
 
 void createTerminator(TypeResults &TR, DiffeGradientUtils *gutils,
                       BasicBlock *oBB, DIFFE_TYPE retType, ReturnType retVal) {
+  ReturnInst *inst = dyn_cast_or_null<ReturnInst>(oBB->getTerminator());
+  // In forward mode we only need to update the return value
+  if (inst == nullptr)
+    return;
 
-  BasicBlock *nBB = cast<BasicBlock>(gutils->getNewFromOriginal(oBB));
+  ReturnInst *newInst = dyn_cast<ReturnInst>(gutils->getNewFromOriginal(inst));
```

Comment:
We should just do a cast rather than dyn_cast here.

Reply:
Yes, but we already checked the original instruction as a return. Should we not have the return here as well?